libs/shareProcessor.js

```diff
@@ -85,6 +85,7 @@ module.exports = function(logger, poolConfig){
             redisCommands.push(['rename', coin + ':shares:roundCurrent', coin + ':shares:round' + shareData.height]);
             redisCommands.push(['rename', coin + ':shares:timesCurrent', coin + ':shares:times' + shareData.height]);
             redisCommands.push(['sadd', coin + ':blocksPending', [shareData.blockHash, shareData.txHash, shareData.height, shareData.worker, dateNow].join(':')]);
+            redisCommands.push(['hset', 'Allblocks', coin + "-" + shareData.height, [shareData.blockHash, shareData.txHash, shareData.height, shareData.worker, dateNow].join(':')]); //used for block stat
```

Comment:
This seems like duplicate data in redis that is not required.

To get the last mined block without this duplicate data being stored in redis.

Since pending/confirmed blocks are sorted in newest first order, the last mined block is always first in the array.
`it.stats.pools[pool].pending.blocks[0]` or `it.stats.pools[pool].confirmed.blocks[0]`

Example
```
var lastblock;
if (it.stats.pools[pool].pending.blocks.length > 0) {
  lastblock = it.stats.pools[pool].pending.blocks[0];
} else {
  lastblock = it.stats.pools[pool].confirmed.blocks[0];
}
```
Some additional checks may be required in case `it.stats.pools[pool].confirmed.blocks` or `it.stats.pools[pool].pending.blocks`  is undefined. (No pending blocks, etc.)